CHANGELOG.md

```diff
@@ -10,12 +10,318 @@ New features:
+
+The special unary `#` syntax for row kinds is still supported, but deprecated and will warn. There is now `Prim.Row :: Type -> Type` which can be used like a normal type constructor.
+
+> Unary '#' syntax for row kinds is deprecated and will be removed in a future release. Use the 'Row' kind instead.
+
+All of these deprecations have suggested fixes in the JSON output, so tools like [`purescript-suggest`](https://github.com/nwolverson/purescript-suggest) (or your IDE plugin) can automatically apply them.
+
+#### Kind Signatures
+
+With PolyKinds, all type-level declarations are generalized.
+
+```purescript
+data Proxy a = Proxy
+```
+
+Previously, this had the `Type`-defaulted kind `Type -> Type`. Now this will be generalized to `forall k. k -> Type`. Such signature can be written with a kind signature declarations, similar to [standalone kind signatures](https://ryanglscott.github.io/2020/01/05/five-benefits-to-using-standalonekindsignatures/) in GHC.
+
+```purescript
+data Proxy :: forall k. k -> Type
+data Proxy a = Proxy
+```
+
+In GHC, all signatures use the `type` prefix, but we reuse the same keyword as the subsequent declaration because we already have `foreign import data` (rather than `foreign import type`) and because it makes things align nicer. Signatures have the same rule as value-level signatures, so they must always be followed by the "real" declaration.
+
+It's better to be explicit about polymorphism by writing signatures. Since we don't really quantify over free type variables, it's also necessary in the case that two poly-kinded arguments must have the same kind. The compiler will warn about missing kind signatures when polymorphic kinds are inferred.
+
+Classes can have signatures too, but they must end with the new `Constraint` kind instead of `Type`. For example, `Prim.Row.Cons` now has the signature:
+
+```purescript
+class Cons :: forall k. Symbol -> k -> Row k -> Row k -> Constraint
+```
+
+### Safe zero-cost coercions
+
+Coercible constraints, based on the [Safe Zero-cost Coercions for Haskell](https://www.microsoft.com/en-us/research/uploads/prod/2018/05/coercible-JFP.pdf) paper (#3351, #3810, #3896, #3873, #3860, #3905, #3893, #3909, #3931, #3906, #3881, #3878, #3937, #3930, #3955, #3927, @lunaris, @rhendric, @kl0tl, @hdgarrood)
+
+`Prim.Coerce.Coercible` is a new compiler-solved class, used to relate types with the same runtime representation. One can use `Safe.Coerce.coerce` (from the new [`safe-coerce`](https://github.com/purescript/purescript-safe-coerce) library) instead of `Unsafe.Coerce.unsafeCoerce` to safely turn a `a` into a `b` when `Coercible a b` holds.
+
+Coercible constraints are solved according to the following rules:
+
+* Coercible constraints are _reflexive_, `Coercible a a` holds, _symmetric_, `Coercible a b` implies `Coercible b a`, and _transitive_, `Coercible a b` and `Coercible b c` imply `Coercible a c`.
```

Comment:
Perhaps this would be a little easier to parse on multiple lines? ie. coercible constraints satisfy these properties:

```markdown
 * _reflexive_:`Coercible a a` holds.
 * _symmetric_: `Coercible a b` implies `Coercible b a`.
 * _transitive_: `Coercible a b` and `Coercible b c` imply `Coercible a c`.
```

Reply:
I agree with the above change.